cviceni7/petrekova07/tuples_and_sets.ipynb

```diff
@@ -0,0 +1,1017 @@
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "529344067295497451\n",
+      "529344067295497451\n",
+      "-4363729961677198915\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Nezapomínejme, že tuple s neměnnými prvky je hashovatelný\n",
+    "tuple_A = (1, 2, 3)\n",
+    "tuple_B = (1, 2, 3)\n",
+    "tuple_C = (1, 2, 4)\n",
+    "\n",
+    "print(hash(tuple_A))\n",
+    "print(hash(tuple_B))\n",
+    "print(hash(tuple_C))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Tuple_A a Tuple_B jsou stejné\n"
+     ]
+    }
+   ],
+   "source": [
+    "# To můžeme využít k porovnání dvou tuple\n",
+    "if hash(tuple_A) == hash(tuple_B):\n",
+    "    print(\"Tuple_A a Tuple_B jsou stejné\")\n",
+    "\n",
+    "if hash(tuple_A) == hash(tuple_C):\n",
+    "    print(\"Tuple_A a Tuple_C jsou stejné\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(1, 2, [3, 4])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Pozor, pokud tuple obsahuje mutable prvky, tak už není hashovatelný\n",
+    "tuple_D = (1, 2, [3, 4])\n",
+    "\n",
+    "#print(hash(tuple_D)) # chyba\n",
+    "print(tuple_D)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Úkol 1.2\n",
+    "Nyní první psaní! Prve si přečtěte kroky. Think before u code.\n",
+    "\n",
+    "1. Vytvořte si tuple se čtyřmi lat, lon souřadnicemi.\n",
+    "2. Pomocí for i smyčky vytiskněte hash každé souřadnice. Každá vytištěná hash musí být stejná jako ta předchozí.\n",
+    "3. Vytiskněte i hash celého tuple.\n",
+    "\n",
+    "Pod buňkou tedy budou čtyři identické hashe + jedna odlišná."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-1647526307627387439\n",
+      "-1647526307627387439\n",
+      "-1647526307627387439\n",
+      "-1647526307627387439\n",
+      "-168995955776474353\n"
+     ]
+    }
+   ],
+   "source": [
+    "#vytvoření tuple o 4 souřadnicích\n",
+    "vnorene_sour=(\n",
+    "    (49.593778, 17.250878),\n",
+    "    (49.593778, 17.250878),\n",
+    "    (49.593778, 17.250878),\n",
+    "    (49.593778, 17.250878)\n",
+    " )\n",
+    "#vytvoření or smyčky pro vytištění hash každé souřadnice/tuple\n",
+    "#print(hash(vnorene_sour[0]))\n",
+    "#print(hash(vnorene_sour[1]))\n",
+    "#print(hash(vnorene_sour[2]))\n",
+    "#print(hash(vnorene_sour[3]))\n",
+    "# vytvoreni cyklu pro vytisknutí hash každé souřadnice\n",
+    "for i in range(len(vnorene_sour)):\n",
+    "    print(hash(vnorene_sour[i]))\n",
```

Comment:
Je to určitě správně! Zde by stačilo ale použít ještě jednodušíí for in smyčky ve stylu

```
for hodnota in vnorene_sour:
    print(hash(hodnota))
```